spec/services/agency_identity_linker_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+describe AgencyIdentityLinker do
+  let(:user) { create(:user) }
+  describe '#link_identity' do
+    before(:each) { init_env(user) }
+
+    it 'links identities from 2 sps' do
+      sp1 = create_identity(user, 'http://localhost:3000', 'UUID1')
+      create_identity(user, 'urn:gov:gsa:openidconnect:test', 'UUID2')
+      AgencyIdentityLinker.new(sp1).link_identity
+      ai = AgencyIdentity.where(user_id: user.id).first
+      expect(ai.uuid).to eq('UUID1')
+    end
+
+    it 'links identity with 1 sp' do
+      sp1 = create_identity(user, 'http://localhost:3000', 'UUID1')
+      AgencyIdentityLinker.new(sp1).link_identity
+      ai = AgencyIdentity.where(user_id: user.id).first
+      expect(ai.uuid).to eq('UUID1')
+    end
+
+    it 'does not link identity without an agency_id' do
+      sp1 = create_identity(user, 'sp:with:no:agency_id', 'UUID1')
+      AgencyIdentityLinker.new(sp1).link_identity
+      ai = AgencyIdentity.where(user_id: user.id).first
+      expect(ai).to eq(nil)
+    end
+
+    it 'returns the existing agency_identity if it exists' do
+      sp1 = create_identity(user, 'http://localhost:3000', 'UUID1')
+      AgencyIdentityLinker.new(sp1).link_identity
+      ai = AgencyIdentity.where(user_id: user.id).first
+      expect(ai.uuid).to eq('UUID1')
+    end
```

Comment:
I would expect these tests to verify the return value of the `#link_identity` method. As written, if you modify the `#link_identity` method to return the String "foo", these tests would still pass.